genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/SetUpJobAction.java

```diff
@@ -504,4 +482,49 @@ private File createJobEnvironmentFile(
         return Collections.unmodifiableMap(env);
     }
 
+    private void cleanupJobDirectory(
+        final File jobDirectory,
+        final CleanupStrategy cleanupStrategy
+    ) throws IOException {
+
+        switch (cleanupStrategy) {
+            case NO_CLEANUP:
+                log.info("Skipping cleanup of job directory: {}", jobDirectory.getAbsolutePath());
+                break;
+
+            case FULL_CLEANUP:
+                log.info("Wiping job directory: {}", jobDirectory.getAbsolutePath());
+                FileSystemUtils.deleteRecursively(jobDirectory);
+                break;
+
+            case DEPENDENCIES_CLEANUP:
+                final RegexRuleSet cleanupWhitelist = RegexRuleSet.buildWhitelist(
+                    (Pattern[]) Lists.newArrayList(
+                        PathUtils.jobClusterDirectoryPath(jobDirectory, ".*"),
+                        PathUtils.jobCommandDirectoryPath(jobDirectory, ".*"),
+                        PathUtils.jobApplicationDirectoryPath(jobDirectory, ".*")
+                    )
+                        .stream()
+                        .map(PathUtils::jobEntityDependenciesPath)
```

Comment:
Would it be better to combine all these separate map statements into one map? I'm not sure if it helps performance so it might be fine to leave it this way for readability.